### DISCUSSION TITLE
How to get currently visible data?

### DISCUSSION THREAD

**Discussion by BarishSarac** [TanStack/table#2312] [General] [answered]
Hi,

I need to get a hold of visible data within the table. Using below code, I can get visible columns data. Like if 40 columns and 50 rows of data there is and only 10 columns and 30 rows of data visible, I get 30 rows of data still with 10 visible columns, with the help of Lodash. How can filter down not visible rows from this data set? I need to do this within TableToolbar element

 ```
let VisibleColumns = columns
    .filter((col) => (col.isVisible === true) & (col.id !== "selection"))
    .map((col) => col.id);
  // Below picks all available data shown on the page
  let VisibleData = data.map((row) => pick(row, VisibleColumns));
```

**Comment by BarishSarac** [comment]
Is there no way to do this?

**Comment by gargroh** [comment, marked as answer]
Try using  `rows` key which is available in table `instance`

**Reply by tannerlinsley** [reply]
For now, you would have to filter each row's cells down to the visible columns in your table, then use the values from those cells.

**Reply by ZeldOcarina** [reply]
Is it still the case? `rows` doesn't seem available on `table`currently!
